### PR TITLE
AF-1028 : [Project Oriented] Standalone perspectives are broken

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/AF-1028

Due to Library changes just the NoContext AuthoringPerspective makes sense now. So removing the old Authoring and renaming NoContext for easier use.

https://github.com/kiegroup/kie-wb-common/pull/1524
https://github.com/kiegroup/kie-wb-distributions/pull/714
https://github.com/kiegroup/kie-docs/pull/735

### CODE REVIEW

**Comment by Rikkola** [comment]
@jhrcek This kills the original AuthoringPerspective and replaces it with the NoContext

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java [hunk 1]
```diff
@@ -1,152 +1,166 @@
...
-    public void setup() {
-        docks.setup( PerspectiveIds.AUTHORING, new DefaultPlaceRequest( "org.kie.guvnor.explorer" ) );
+    public void init() {
+        explorerMode = ( ( Window.Location.getParameterMap().containsKey( "explorer_mode" ) ) ? Window.Location.getParameterMap().get( "explorer_mode" ).get( 0 ) : "" ).trim();
+        projectPathString = ( ( ( Window.Location.getParameterMap().containsKey( "path" ) ) ? Window.Location.getParameterMap().get( "path" ).get( 0 ) : "" ) ).trim();
+        projectEditorDisableBuild = Window.Location.getParameterMap().containsKey("no_build");
+
+        final PlaceRequest placeRequest = generateProjectExplorerPlaceRequest();
+
+        docks.setup("AuthoringPerspective", placeRequest);
```
Comment (jhrcek) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java]: Could you please reuse `PerspectiveIds.AUTHORING` here?

File: kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/DroolsAuthoringPerspective.java [hunk 2]
```diff
@@ -48,106 +53,114 @@
...
-    public void setup() {
-        docks.setup( PerspectiveIds.AUTHORING, new DefaultPlaceRequest( "org.kie.guvnor.explorer" ) );
+    public void init() {
+        explorerMode = ( ( Window.Location.getParameterMap().containsKey( "explorer_mode" ) ) ? Window.Location.getParameterMap().get( "explorer_mode" ).get( 0 ) : "" ).trim();
+        projectPathString = ( ( ( Window.Location.getParameterMap().containsKey( "path" ) ) ? Window.Location.getParameterMap().get( "path" ).get( 0 ) : "" ) ).trim();
+        projectEditorDisableBuild = Window.Location.getParameterMap().containsKey( "no_build" );
+
+        final PlaceRequest placeRequest = generateProjectExplorerPlaceRequest();
+
+        docks.setup("AuthoringPerspective", placeRequest);
```
Comment (jhrcek) [inline kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/DroolsAuthoringPerspective.java]: And here as well `PerspectiveIds.AUTHORING`. It's in scope, so let's reuse it..

**Comment by jhrcek** [comment]
@Rikkola thanks for reintroducing the integration test. Let jenkins retest this a few times to verify it's stable..

**Comment by Rikkola** [comment]
sure, jenkins please retest this

**Comment by jhrcek** [comment]
2 passes in the row, looks stable. Last one :) 
jenkins retest this

**Comment by Rikkola** [comment]
@jhrcek Nice. I'll open the follow up ticket for this.

**Comment by Rikkola** [comment]
https://issues.jboss.org/browse/AF-1115 In case anyone wants to track the ticket.

**Comment by jhrcek** [comment]
@Rikkola Sorry for late notice, but I'm updating our tests for standalone perspective and I discovered one thing that doesn't work: I can't get project explorer to open as you describe here (taken from updated kie-docs).

> | http://localhost:8080/kie-wb/kie-wb.jsp?standalone&perspective=AuthoringPerspective&path=default://master@myteam/Mortgages/src/main/java/mortgages/mortgages/Applicant.java
| Opens the given file with the Project Explorer. Giving the user the possibility to browse the content of the project where the given file is and even open other files from the project.

This kind of url does open the editor within standalone AuthoringPerspective, but there's no project explorer to be found anywhere.. Should I file yet another jira for that?

**Comment by Rikkola** [comment]
@jhrcek I need to check that. I copy pasted the URLs into the docs after I tested them.

**Comment by Rikkola** [comment]
Finally back on this! Haven't forgotten! ;)

**Comment by Rikkola** [comment]
@jhrcek So worked for me. Not sure what could cause the inconsistency.

**Comment by jhrcek** [comment]
Jenkins execute full downstream build

**Comment by Rikkola** [comment]
So kie-drools-wb.war works, but kie-wb.war is broken. To me changes look identical, the codes on this area are redundant so trying to figure out what could be different.

**Comment by Rikkola** [comment]
Maybe it is worth getting this in as it is. It fixes kie-drools-wb and kie-wb partially. I'm still working on kie-wb.war, but just going from one dead end to another.

I'll still continue fixing kie-wb.war Project Explorer issue.

**Comment by Rikkola** [comment]
@jhrcek Actually that is not, but that might be related. Good thing that came up.

**Comment by jhrcek** [comment]
@Rikkola is there another jira to track that issue? I'd like to add myself as a watcher..

**Comment by Rikkola** [comment]
@jhrcek I just made this to track the additional work. 
https://issues.jboss.org/browse/DROOLS-2535

**Comment by mareknovotny** [comment]
have you got an agreement to merge this even tests fail?

**Comment by jhrcek** [comment]
Let's just double check it still compiles. It's been a long time since this saw some CI..  jenkins retest this

**Comment by jhrcek** [comment]
jenkins retest this

**Comment by jhrcek** [comment]
Looks like kie-wb war deployment timeout in rest test (failed to start within 90 seconds, but I see no errors in server console..). 
jenkins retest this

**Comment by jhrcek** [comment]
@Rikkola @manstis There's 1 deterministic selenium test failure cause by @porcelli 's [change](https://github.com/kiegroup/kie-wb-common/commit/e4ecb9ef2547e6ce916984eb4790f5f01a5b318c) - I'll create PR with fix to that test later today. Let's merge this.

**Pull request merged**